cmd/controller-manager/app/controller_manager.go

```diff
@@ -170,40 +176,44 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 		panic("unreachable")
 	}
 
-	// TODO: leader election for this controller-manager
-	// id, err := os.Hostname()
-	// if err != nil {
-	// 	return err
-	// }
-	// rl := resourcelock.EndpointsLock{
-	// 	EndpointsMeta: v1.ObjectMeta{
-	// 		Namespace: "kube-system",
-	// 		Name:      "kube-controller-manager",
-	// 	},
-	// 	Client: leaderElectionClient,
-	// 	LockConfig: resourcelock.ResourceLockConfig{
-	// 		Identity:      id,
-	// 		EventRecorder: recorder,
-	// 	},
-	// }
+	if !controllerManagerOptions.LeaderElection.LeaderElect {
+		run(nil)
```

Comment:
I was a little concerned that this could lead to a segfault due to nil pointer dereference farther on down the line, but I've put it to the test in the go playground and it seems safe. All the same, it might be better not to let anyone else reading this code get distracted by that same worry. `run(make(<-chan (struct{})))` seems more clear.